File: app/src/main/java/com/neocaptainnemo/testing/service/DistanceFormatter.java
package com.neocaptainnemo.testing.service;

import android.content.Context;

import com.neocaptainnemo.testing.R;

import javax.inject.Inject;

public class DistanceFormatter {
    private Context context;

    @Inject
    DistanceFormatter(Context context) {
        this.context = context;
    }

    public String format(double meters) {
        if (meters < 1000.0) {
            return context.getString(R.string.distance_m, meters);
        } else {
            return context.getString(R.string.distance_km, meters / 1000.0);

        }

    }
}

File: app/src/main/java/com/neocaptainnemo/atms/service/DistanceFormatter.kt
package com.neocaptainnemo.atms.service

import android.content.Context

import com.neocaptainnemo.atms.R

import javax.inject.Inject
import javax.inject.Singleton

class DistanceFormatter @Inject constructor(private val context: Context) {

    fun format(meters: Double): String {
        return if (meters < 1000.0) {
            context.getString(R.string.distance_m, meters)
        } else {
            context.getString(R.string.distance_km, meters / 1000.0)

        }

    }
}
